staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go

```diff
@@ -245,11 +274,16 @@ func (w *deferredResponseWriter) Write(p []byte) (n int, err error) {
 	return w.w.Write(p)
 }
 
-func (w *deferredResponseWriter) Close() error {
+func (w *deferredResponseWriter) Close() (err error) {
 	if !w.hasWritten {
-		return nil
+		if !w.hasBuffered {
+			return nil
+		}
+		_, err := w.unbufferedWrite(w.buffer)
+		w.buffer = nil
+		return err
```

Comment:
returning here is not correct... we want to write the buffer and then attempt to close... if this passed tests then we need a test covering this bug

Reply:
(calling unbufferedWrite will flip hasWritten from false to true)

Reply:
ah, Close() was only doing something in the gzip case... I thought it was passing through to call w.w.Close() in all cases

Reply:
I just realized `io.Writer` doesn't even provide the Close()